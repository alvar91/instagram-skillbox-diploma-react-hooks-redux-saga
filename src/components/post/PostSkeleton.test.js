// Core
import React from "react";
import { configure, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import PostSkeleton from "./PostSkeleton";

configure({ adapter: new Adapter() });

/**
 * Setup function for PostSkeleton component.
 * @param {object} props
 * @returns {ReactRenderer}
 */
const setupTestRenderer = () => {
  return mount(<PostSkeleton />);
};

describe("PostSkeleton", () => {
  let testRenderer;
  let component;

  beforeAll(() => {
    testRenderer = setupTestRenderer();
    component = testRenderer.find(PostSkeleton);
  });

  it("PostSkeleton should correctly renders markup", () => {
    expect(shallowToJson(testRenderer)).toMatchSnapshot();
  });
});
