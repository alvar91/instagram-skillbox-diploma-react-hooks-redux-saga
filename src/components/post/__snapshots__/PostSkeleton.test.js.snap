// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSkeleton PostSkeleton should correctly renders markup 1`] = `
<PostSkeleton>
  <div
    className="makeStyles-container-1"
    style={
      Object {
        "gridTemplateColumns": false,
      }
    }
  >
    <div
      className="makeStyles-mediaSkeleton-7"
    />
    <div>
      <div
        className="makeStyles-headerSkeleton-2"
      >
        <div
          className="makeStyles-avatarSkeleton-3"
        />
        <div
          className="makeStyles-headerTextSkeleton-4"
        >
          <div
            className="makeStyles-primaryTextSkeleton-5"
          />
          <div
            className="makeStyles-secondaryTextSkeleton-6"
          />
        </div>
      </div>
    </div>
  </div>
</PostSkeleton>
`;
